components/mediation/mediators/dataservices-mediator/org.wso2.micro.integrator.mediator.dataservice/src/main/java/org/wso2/micro/integrator/mediator/dataservice/DataServiceCallMediatorFactory.java

```diff
@@ -0,0 +1,202 @@
+ *          &lt;operation name=""&gt;
+ *                  &lt;param name="string"&gt; &lt;param/&gt;
+ *          &lt;/operation&gt;
+ *      &lt;/operations&gt;
+ *      &lt;target  type="envelope"|"property" name="string"/&gt;
+ * &lt;/dataServicesCall&gt;
+ * </pre>
+ */
+
+public class DataServiceCallMediatorFactory extends AbstractMediatorFactory {
+
+    private static final QName DSSCALL_Q =
+            new QName(SynapseConstants.SYNAPSE_NAMESPACE, DataServiceCallMediatorConstants.DATA_SERVICES_CALL);
+    private static final QName TARGET_Q = new QName(
+            XMLConfigConstants.SYNAPSE_NAMESPACE, DataServiceCallMediatorConstants.TARGET);
+    private static final QName NAME_Q = new QName(DataServiceCallMediatorConstants.NAME);
+    private static final QName TYPE_Q = new QName(DataServiceCallMediatorConstants.TYPE);
+
+
+    public Mediator createSpecificMediator(OMElement elem, Properties properties) {
+
+        DataServiceCallMediator mediator = new DataServiceCallMediator();
+        processAuditStatus(mediator, elem);
+        String dsName = elem.getAttributeValue(new QName(DataServiceCallMediatorConstants.SERVICE_NAME));
+        if (dsName == null) {
+            handleException("The 'serviceName' attribute in 'dataServicesCall' element  is missing " +
+                    "in the configuration.");
+        }
+        mediator.setDsName(dsName);
+        OMElement operationsTypeElement = elem.getFirstElement();
+        if (!operationsTypeElement.getLocalName().equals(DataServiceCallMediatorConstants.OPERATIONS)) {
+            handleException("The 'operations' element in 'dataServicesCall' element  is missing in the configuration.");
+        }
+        String operationType = operationsTypeElement.getAttributeValue(new QName(DataServiceCallMediatorConstants.TYPE));
+        if (operationType == null) {
+            handleException("The 'type' attribute in 'operations' element  is missing in the configuration.");
+        }
+        OperationsType operationsType = OperationsType.valueOf(operationType.toUpperCase());
+        List operationList = extractOperations(operationsTypeElement, mediator);
+        Operations operations = mediator.new Operations(operationsType, operationList);
+        mediator.setOperations(operations);
+
+        OMElement targetElement = elem.getFirstChildWithName(TARGET_Q);
+        if (targetElement != null) {
+            OMAttribute typeAtr = targetElement.getAttribute(TYPE_Q);
+            if (typeAtr != null) {
+                String type = typeAtr.getAttributeValue();
+                mediator.setTargetType(type);
+                if (type.equals(DataServiceCallMediatorConstants.PROPERTY)) {
+                    OMAttribute propertyAtr = targetElement.getAttribute(NAME_Q);
+                    if (propertyAtr != null) {
+                        if (!propertyAtr.getAttributeValue().isEmpty()) {
+                            String propertyName = propertyAtr.getAttributeValue();
```

Comment:
is it possible to make this allow dynamic property names as we have in property mediator now

Reply:
+1. Will add

Reply:
https://github.com/wso2/micro-integrator/issues/1726